Buildings/Experimental/DHC/EnergyTransferStations/Combined/Generation5/Controls/LimPIDEnable.mo

```diff
@@ -0,0 +1,106 @@
+  parameter Real y_reset=yMin
+    "Value to which the controller output is reset if the boolean trigger has a rising edge";
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput u_s
+    "Connector of setpoint input signal"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}}),iconTransformation(extent={{-140,-20},{-100,20}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput u_m
+    "Connector of measurement input signal"
+    annotation (Placement(transformation(origin={0,-120},extent={{20,-20},{-20,20}},rotation=270),iconTransformation(extent={{20,-20},{-20,20}},rotation=270,origin={0,-120})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput y
+    "Connector of actuator output signal"
+    annotation (Placement(transformation(extent={{100,-20},{140,20}}),iconTransformation(extent={{100,-20},{140,20}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uEna
+    "Enable signal"
+    annotation (Placement(transformation(extent={{-20,-20},{20,20}},rotation=90,origin={-60,-120}),iconTransformation(extent={{-20,-20},{20,20}},rotation=90,origin={-40,-120})));
+  Buildings.Controls.OBC.CDL.Continuous.PIDWithReset conPID(
+    final k=k,
+    final Ti=Ti,
+    final Td=Td,
+    final controllerType=controllerType,
+    final yMin=yMin,
+    final yMax=yMax,
+    final reverseActing=reverseActing)
+    annotation (Placement(transformation(extent={{-10,-10},{10,10}})));
+  Buildings.Controls.OBC.CDL.Logical.Switch swi
+    annotation (Placement(transformation(extent={{-40,-10},{-20,10}})));
+  Buildings.Controls.OBC.CDL.Logical.Switch swi1
+    annotation (Placement(transformation(extent={{72,-10},{92,10}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant valDis(
+    final k=0)
+    "Value when disabled"
+    annotation (Placement(transformation(extent={{30,-50},{50,-30}})));
+equation
+  connect(conPID.u_s,swi.y)
+    annotation (Line(points={{-12,0},{-18,0}},color={0,0,127}));
+  connect(uEna,swi.u2)
+    annotation (Line(points={{-60,-120},{-60,-21.5625},{-60,0},{-42,0}},color={255,0,255}));
+  connect(u_s,swi.u1)
+    annotation (Line(points={{-120,0},{-80,0},{-80,8},{-42,8}},color={0,0,127}));
+  connect(u_m,swi.u3)
+    annotation (Line(points={{0,-120},{0,-80},{-50,-80},{-50,-8},{-42,-8}},color={0,0,127}));
+  connect(uEna,conPID.trigger)
+    annotation (Line(points={{-60,-120},{-60,-20},{-6,-20},{-6,-12}},color={255,0,255}));
+  connect(u_m,conPID.u_m)
+    annotation (Line(points={{0,-120},{0,-12}},color={0,0,127}));
+  connect(conPID.y,swi1.u1)
+    annotation (Line(points={{12,0},{40,0},{40,8},{70,8}},color={0,0,127}));
+  connect(swi1.y,y)
+    annotation (Line(points={{94,0},{120,0}},color={0,0,127}));
+  connect(uEna,swi1.u2)
+    annotation (Line(points={{-60,-120},{-60,-20},{60,-20},{60,0},{70,0}},color={255,0,255}));
+  connect(valDis.y,swi1.u3)
+    annotation (Line(points={{52,-40},{64,-40},{64,-8},{70,-8}},color={0,0,127}));
+  annotation (
```

Comment:
add `defaultComponentName`

Reply:
Addressed (`conPID`).
I also renamed the class `PIDWithEnable` for consistency with the CDL PID block recent renaming.